database-commons/src/main/java/io/cdap/plugin/db/batch/action/AbstractDBArgumentSetter.java

```diff
@@ -0,0 +1,140 @@
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    FailureCollector collector = stageConfigurer.getFailureCollector();
+    config.validate(collector);
+    try {
+      processArguments(driverClass, collector, null);
+    } catch (SQLException e) {
+      collector.addFailure("SQL error while executing query: " + e.getMessage(), null)
+        .withStacktrace(e.getStackTrace());
+    } catch (IllegalAccessException | InstantiationException e) {
+      collector.addFailure("Unable to instantiate JDBC driver: " + e.getMessage(), null)
+        .withStacktrace(e.getStackTrace());
+    } catch (Exception e) {
+      collector.addFailure(e.getMessage(), null).withStacktrace(e.getStackTrace());
+    }
+  }
+
+  /**
+   * Creates connection to database. Reads row from database based on selection conditions and
+   * depending on whether settable arguments is provided or not set the argument from row columns.
+   *
+   * @param driverClass       {@link Class<? extends Driver>}
+   * @param failureCollector  {@link FailureCollector}
+   * @param settableArguments {@link SettableArguments}
+   * @throws SQLException           is raised when there is sql related exception
+   * @throws IllegalAccessException is raised when there is access related exception
+   * @throws InstantiationException is raised when there is class/driver issue
+   */
+  private void processArguments(Class<? extends Driver> driverClass,
+                                FailureCollector failureCollector, SettableArguments settableArguments)
+    throws SQLException, IllegalAccessException, InstantiationException {
+    DriverCleanup driverCleanup;
+
+    driverCleanup = DBUtils.ensureJDBCDriverIsAvailable(driverClass, config.getConnectionString(),
+                                                        config.jdbcPluginName);
+    Properties connectionProperties = new Properties();
+    connectionProperties.putAll(config.getConnectionArguments());
+    try {
+      Connection connection = DriverManager
+        .getConnection(config.getConnectionString(), connectionProperties);
+      Statement statement = connection.createStatement();
+      ResultSet resultSet = statement.executeQuery(config.getQuery());
+      boolean hasRecord = resultSet.next();
+      if (!hasRecord) {
+        failureCollector.addFailure("No record found.",
+                                    "The argument selection conditions must match only one record.");
+        return;
+      }
+      if (settableArguments != null) {
+        setArguments(resultSet, failureCollector, settableArguments);
+      }
+      if (resultSet.next()) {
+        failureCollector
+          .addFailure("More than one records found.",
+                      "The argument selection conditions must match only one record.");
+      }
+    } catch (SQLException e) {
+      throw new SQLException(e.getMessage(), e.getSQLState(), e.getErrorCode());
```

Comment:
why do we catch the sqlexception and throw the same exception? It should throw same `SQLException`

Reply:
> a few comments. Fix and squash the commits

@CuriousVini done.